demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -632,7 +641,28 @@ static int connectToServerWithBackoffRetries( NetworkContext_t * pNetworkContext
                                          TRANSPORT_SEND_RECV_TIMEOUT_MS,
                                          TRANSPORT_SEND_RECV_TIMEOUT_MS );
 
-        if( opensslStatus != OPENSSL_SUCCESS )
+        if( opensslStatus == OPENSSL_SUCCESS )
+        {
+            /* A clean MQTT session needs to be created, if there is no session saved
+             * in this MQTT client. */
+            createCleanSession = ( *pClientSessionPresent == true ) ? false : true;
+
+            /* Sends an MQTT Connect packet using the established TLS session,
+             * then waits for connection acknowledgment (CONNACK) packet. */
+            returnStatus = establishMqttSession( pMqttContext, createCleanSession, pBrokerSessionPresent );
+
+            if( returnStatus == EXIT_FAILURE )
+            {
+                /* End TLS session, then close TCP connection. */
+                ( void ) Openssl_Disconnect( pNetworkContext );
+            }
+            else
+            {
+                *pClientSessionPresent = true;
+            }
+        }
+
+        if( ( opensslStatus != OPENSSL_SUCCESS ) || ( returnStatus == EXIT_FAILURE ) )
```

Comment:
Actually, the `while` loop on 683 needs to be updated as well, and/or maybe this should just be changed to only check `returnStatus`, and we initialize that one to `EXIT_FAILURE`. That way both the TLS and MQTT connection would have to succeed before `returnStatus` is `EXIT_SUCCESS`

Reply:
I have update both the while and if condition to ( returnStatus == EXIT_FAILURE )